### ISSUE TITLE
making a better divertor for ball reactor

### ISSUE THREAD

**Issue by Shimwell** [ukaea/paramak#515]
Currently we have a diverter envelop that connects to the blanket

It works reasonably for divertor_radial_thickness which result in it being above the plasma but for lower divertor_radial_thickness it has a mismatch with the blanket connection

Here is an example of 26, 50 and 100 for the divertor radial thickness

```python
import paramak
my_reactor = paramak.BallReactor(
        inner_bore_radial_thickness=10,
        inboard_tf_leg_radial_thickness=30,
        center_column_shield_radial_thickness=60,
        divertor_radial_thickness=100,
        inner_plasma_gap_radial_thickness=30,
        plasma_radial_thickness=300,
        outer_plasma_gap_radial_thickness=30,
        firstwall_radial_thickness=30,
        blanket_radial_thickness=50,
        blanket_rear_wall_radial_thickness=30,
        elongation=2,
        triangularity=0.55,
        number_of_tf_coils=16,
        rotation_angle=180,
        pf_coil_radial_thicknesses=[50, 50, 50, 50],
        pf_coil_vertical_thicknesses=[50, 50, 50, 50],
        pf_coil_to_rear_blanket_radial_gap=50,
        pf_coil_to_tf_coil_radial_gap=50,
        outboard_tf_coil_radial_thickness=100,
        outboard_tf_coil_poloidal_thickness=50
)
my_reactor.export_stp()
```

```python
import paramak
my_reactor = paramak.BallReactor(
        inner_bore_radial_thickness=10,
        inboard_tf_leg_radial_thickness=30,
        center_column_shield_radial_thickness=60,
        divertor_radial_thickness=50,
        inner_plasma_gap_radial_thickness=30,
        plasma_radial_thickness=300,
        outer_plasma_gap_radial_thickness=30,
        firstwall_radial_thickness=30,
        blanket_radial_thickness=50,
        blanket_rear_wall_radial_thickness=30,
        elongation=2,
        triangularity=0.55,
        number_of_tf_coils=16,
        rotation_angle=180,
        pf_coil_radial_thicknesses=[50, 50, 50, 50],
        pf_coil_vertical_thicknesses=[50, 50, 50, 50],
        pf_coil_to_rear_blanket_radial_gap=50,
        pf_coil_to_tf_coil_radial_gap=50,
        outboard_tf_coil_radial_thickness=100,
        outboard_tf_coil_poloidal_thickness=50
)
my_reactor.export_stp()
```

```python
import paramak
my_reactor = paramak.BallReactor(
        inner_bore_radial_thickness=10,
        inboard_tf_leg_radial_thickness=30,
        center_column_shield_radial_thickness=60,
        divertor_radial_thickness=26,
        inner_plasma_gap_radial_thickness=30,
        plasma_radial_thickness=300,
        outer_plasma_gap_radial_thickness=30,
        firstwall_radial_thickness=30,
        blanket_radial_thickness=50,
        blanket_rear_wall_radial_thickness=30,
        elongation=2,
        triangularity=0.55,
        number_of_tf_coils=16,
        rotation_angle=180,
        pf_coil_radial_thicknesses=[50, 50, 50, 50],
        pf_coil_vertical_thicknesses=[50, 50, 50, 50],
        pf_coil_to_rear_blanket_radial_gap=50,
        pf_coil_to_tf_coil_radial_gap=50,
        outboard_tf_coil_radial_thickness=100,
        outboard_tf_coil_poloidal_thickness=50
)
my_reactor.export_stp()
```

![Screenshot from 2020-11-17 10-17-15](https://user-images.githubusercontent.com/8583900/99377862-a4e06a00-28be-11eb-9e88-cbea19b6a711.png)

![Screenshot from 2020-11-17 10-16-06](https://user-images.githubusercontent.com/8583900/99377856-a447d380-28be-11eb-8c37-438ae1f6b798.png)

![Screenshot from 2020-11-17 10-15-08](https://user-images.githubusercontent.com/8583900/99377855-a3af3d00-28be-11eb-8934-9d9e777567cd.png)

**Comment by Shimwell** [comment]
I think we need a minimal example of this intersection between a cylinder shape and a spline shape to show the resulting geometry is not where it should be. Then perhaps as CQ devs

**Comment by RemDelaporteMathurin** [comment]
Fixed in #516
